### PR TITLE
RAC: `Modal` &  `Dialog`

### PR DESCRIPTION
**Pull request by OsamaAbdellateef**

### CODE REVIEW

**Comment by sebald** [comment]
Is this ready or still a draft? :)

**Comment by OsamaAbdellateef** [comment]
> Is this ready or still a draft? :)

I can say it is ready

File: packages/components/src/Dialog/Dialog.stories.tsx [hunk 1]
```diff
@@ -40,52 +41,48 @@ const meta = {
...
-            )}
-          </Dialog>
-        </Dialog.Trigger>
-      </>
+      <Dialog.Trigger {...args}>
+        <Button variant="primary">Open</Button>
+        <Dialog>
+          {({ close }) => (
+            <>
+              <Headline>Please log into account</Headline>
```
Comment (sebald) [inline packages/components/src/Dialog/Dialog.stories.tsx:71]: Is the slot missing here? Maybe we add a "Dialog.Headline" that anyways had the right slot?

File: packages/components/src/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -1,52 +1,28 @@
...
-  const { close } = useDialogContext();
-
-  const { buttonProps } = useButton(
-    {
-      onPress: () => close?.(),
-    },
-    ref
-  );
-
+const CloseButton = ({ className, close }: CloseButtonProps) => {
```
Comment (sebald) [inline packages/components/src/Dialog/Dialog.tsx:17]: I think this should use the dialog context and not accept a close method, like here:

https://react-spectrum.adobe.com/react-aria/Dialog.html#state

File: packages/components/src/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -60,79 +36,50 @@ const CloseButton = ({ className }: CloseButtonProps) => {
...
-  const ref = useRef(null);
-  const { close } = useDialogContext();
-  const { dialogProps, titleProps } = useDialog(props, ref);
-
   const classNames = useClassNames({ component: 'Dialog', variant, size });
+  let state = useContext(OverlayTriggerStateContext);
 
+  let children = props.children;
+
+  if (typeof children === 'function') {
```
Comment (sebald) [inline packages/components/src/Dialog/Dialog.tsx:62]: Do we still need this? Does the RAC dialog handle it for us?

File: packages/components/src/Dialog/_Dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+export interface DialogProps extends AriaDialogProps {
+  children?: ReactNode | ((props: DialogChildProps) => ReactNode);
+  variant?: string;
+  size?: string;
+  closeButton?: boolean;
+}
+
+// Component
+// ---------------
+export const Dialog = ({
```
Comment (sebald) [inline packages/components/src/Dialog/_Dialog.tsx:112]: This is the old Dialog, right? :-D

File: packages/components/src/Dialog/Dialog.tsx [hunk 5]
```diff
@@ -60,79 +37,57 @@ const CloseButton = ({ className }: CloseButtonProps) => {
...
-      {closeButton && <CloseButton className={classNames.closeButton} />}
-      {typeof children === 'function'
-        ? children({ close, titleProps })
-        : props['aria-labelledby']
-        ? children
-        : addTitleProps(children, titleProps)}
-    </div>
+    <Dialog
+      {...props}
+      className={cn(classNames.container, 'relative outline-none')}
```
Comment (sebald) [inline packages/components/src/Dialog/Dialog.tsx:79]: should be the other way around so that the `container` can override the defaults :)

**Comment by sebald** [comment]
@OsamaAbdellateef this is awesome work! 💪 

@sarahgm @aromko can one of you also make a review? This is a lot of stuff at once. so I didn't miss anything 🙈

File: packages/components/src/Dialog/Dialog.tsx (hunk 5 above)
Comment (sebald) [inline packages/components/src/Dialog/Dialog.tsx:79]:
```suggestion
      className={cn('relative outline-none', classNames.container)}
```

File: packages/components/src/Dialog/Dialog.tsx [hunk 6]
```diff
@@ -60,79 +36,56 @@ const CloseButton = ({ className }: CloseButtonProps) => {
...
 // Props
 // ---------------
-export interface DialogChildProps {
-  close: DialogContextProps['close'];
-  titleProps: HTMLAttributes<HTMLElement>;
-}
-
-export interface DialogProps extends AriaDialogProps {
-  children?: ReactNode | ((props: DialogChildProps) => ReactNode);
+export interface DialogProps extends RAC.DialogProps {
```
Comment (sarahgm) [inline packages/components/src/Dialog/Dialog.tsx:49]: can you remove the `className` and `style` from here? We called it removedProps @OsamaAbdellateef

**Review by sarahgm — APPROVED** [review]

**Pull request merged**